src/plugins/lexer/mod.ts

```diff
@@ -0,0 +1,390 @@
+  };
+  // stores current built token
+  token: string = "";
+  // stores current lexical tokens
+  tokens: Token[] = [];
+  // stores mode lexer is interpreting in
+  mode: Mode = Mode.Token;
+
+  // optional initiator for .mtl source
+  constructor(input?: string) {
+    if (input !== undefined) {
+      this.source = input.replace(/(\r\n|\n|\r)/gm, "&eol") + "&eof";
+    }
+  }
+
+  // loads new .mtl source, avoiding reinitializing class
+  load(input: string) {
+    this.source = input.replace(/(\r\n|\n|\r)/gm, "&eol") + "&eof";
+    return this;
+  }
+
+  // starts generation of lexical tokens
+  get generate() {
+    this.position = -1;
+    this.virtual_position = -1;
+    this.token = "";
+    this.trace = {
+      line: 1,
+      column: [0, 0],
+    };
+    this.tokens = [];
+    this.mode = Mode.Token;
+
+    // basic character loop
+    while ((this.source.length) > this.position) {
+      this.comprehend(this.next());
+    }
+    return this.tokens;
+  }
+
+  // used to build an understanding of individual tokens through individual characters
+  comprehend(character: string): void {
+    // if comprehending new token, store initial `character` location for `trace`
+    if (
+      (this.token.length === 0) && (this.mode !== Mode.String) &&
+      (this.mode !== Mode.Compiler)
+    ) {
+      this.trace.column[0] = this.virtual_position;
+    }
+    // verify mode lexer is in
+    switch (this.mode) {
+      case Mode.Token: {
+        // switch to string mode
+        if ((character === '"') && (this.token.length === 0)) {
+          this.mode = Mode.String;
+          break;
+
+          // presume token is finished; generate
+        } else if (character === " ") {
+          if (this.token.length === 0) return;
+          this.checkToken();
+
+          // switch to compiler call mode
+        } else if (character === "&") {
+          this.mode = Mode.Compiler;
+          this.checkToken();
+
+          // check if current character is an operator
+        } else if (this.isOperator(character) === true) {
+          // handles operator being beside another token
+          this.checkToken();
+          this.token += character;
+          // makes sure column position is accurate for operator
+          this.trace.column[0] = this.virtual_position;
+          this.checkToken(1);
+        } else {
+          // if no conditions met, keep building token
+          this.token += character;
+        }
+        break;
+      }
+
+      case Mode.String: {
+        // check if end of string was reached; switch back to `token` mode
+        if ((character === '"')) {
+          this.mode = Mode.Token;
+          this.trace.column[1] = this.virtual_position;
+          this.createToken("string", this.token);
+
+          // check if next character should be escaped
+        } else if (character === "\\") {
+          this.token += this.next();
+
+          // if no conditions met, keep building string
+        } else {
```

Comment:
Had issues replicating on my side.. can you post the code example for when this issue occurred?

_From my testing:_
testing.mtl
```
const my_lucky_phrase_2: string { "this is \"escaped\"" }
```

output:
```
[
  { type: "keyw", value: "const", trace: { line: 1, column: [ 0, 4 ] } },
  { type: "ident", value: "my_lucky_phrase_2", trace: { line: 1, column: [ 6, 22 ] } },
  { type: "colon", value: ":", trace: { line: 1, column: [ 23, 23 ] } },
  { type: "type", value: "string", trace: { line: 1, column: [ 25, 30 ] } },
  { type: "lcb", value: "{", trace: { line: 1, column: [ 32, 32 ] } },
  {
    type: "string",
    value: 'this is "escaped"',
    trace: { line: 1, column: [ 34, 54 ] }
  },
  { type: "rcb", value: "}", trace: { line: 1, column: [ 56, 56 ] } },
  { type: "eof", value: "*", trace: { line: 1, column: [ -1, -1 ] } }
]
```

Reply:
I think you made some changes on your local branch (perhaps while fixing another suggestion) that fixed this bug. In the current version of this PR:

<img width="701" alt="Screenshot 2022-10-21 at 8 15 26 AM" src="https://user-images.githubusercontent.com/47159695/197193455-cf3a65a8-c17b-4d3d-af5e-29e8fd643218.png">


If this is totally fixed (do some extra testing to be sure) on your next commit, this can be marked as resolved.

Reply:
NOTE: issue has been solved; was due to JS producing unpredicted behaviour with template literals, which is avoided by the Lexer as it only processes the built-in string object. Thanks JS 🪦

For future reference, I should look to see if TypeScript has a type that only takes in the string obect.